libsolidity/analysis/OverrideChecker.cpp

```diff
@@ -69,32 +86,259 @@ vector<ASTPointer<UserDefinedTypeName>> sortByContract(vector<ASTPointer<UserDef
+		[&](ModifierDefinition const* _item) { return _item->virtualSemantics(); },
+		[&](VariableDeclaration const*) { return false; }
+	}, m_item);
+}
+
+Token OverrideProxy::functionKind() const
+{
+	return std::visit(GenericVisitor{
+		[&](FunctionDefinition const* _item) { return _item->kind(); },
+		[&](ModifierDefinition const*) { return Token::Function; },
+		[&](VariableDeclaration const*) { return Token::Function; }
+	}, m_item);
+}
+
+FunctionType const* OverrideProxy::functionType() const
+{
+	return std::visit(GenericVisitor{
+		[&](FunctionDefinition const* _item) { return FunctionType(*_item).asCallableFunction(false); },
+		[&](VariableDeclaration const* _item) { return FunctionType(*_item).asCallableFunction(false); },
+		[&](ModifierDefinition const*) -> FunctionType const* { solAssert(false, "Requested function type of modifier."); return nullptr; }
+	}, m_item);
+}
+
+ModifierType const* OverrideProxy::modifierType() const
+{
+	return std::visit(GenericVisitor{
+		[&](FunctionDefinition const*) -> ModifierType const* { solAssert(false, "Requested modifier type of function."); return nullptr; },
+		[&](VariableDeclaration const*) -> ModifierType const* { solAssert(false, "Requested modifier type of variable."); return nullptr; },
+		[&](ModifierDefinition const* _modifier) -> ModifierType const* { return TypeProvider::modifier(*_modifier); }
+	}, m_item);
+}
+
+SourceLocation const& OverrideProxy::location() const
+{
+	return std::visit(GenericVisitor{
+		[&](auto const* _item) -> SourceLocation const& { return _item->location(); }
+	}, m_item);
+}
+
+string OverrideProxy::astNodeName() const
+{
+	return std::visit(GenericVisitor{
+		[&](FunctionDefinition const*) { return "function"; },
+		[&](ModifierDefinition const*) { return "modifier"; },
+		[&](VariableDeclaration const*) { return "public state variable"; },
+	}, m_item);
+}
+
+string OverrideProxy::astNodeNameCapitalized() const
+{
+	return std::visit(GenericVisitor{
+		[&](FunctionDefinition const*) { return "Function"; },
+		[&](ModifierDefinition const*) { return "Modifier"; },
+		[&](VariableDeclaration const*) { return "Public state variable"; },
+	}, m_item);
+}
+
+string OverrideProxy::distinguishingProperty() const
+{
+	return std::visit(GenericVisitor{
+		[&](FunctionDefinition const*) { return "name and parameter types"; },
+		[&](ModifierDefinition const*) { return "name"; },
+		[&](VariableDeclaration const*) { return "name and parameter types"; },
+	}, m_item);
+}
+
+bool OverrideProxy::unimplemented() const
+{
+	return std::visit(GenericVisitor{
+		[&](FunctionDefinition const* _item) { return !_item->isImplemented(); },
+		[&](ModifierDefinition const*) { return false; },
+		[&](VariableDeclaration const*) { return false; }
+	}, m_item);
+}
+
+bool OverrideProxy::OverrideComparator::operator<(OverrideComparator const& _other) const
+{
+	if (name != _other.name)
+		return name < _other.name;
```

Comment:
Why not `false`?

Reply:
I don't understand - it is a less-than-comparison.

Reply:
Why not `return false`. since the name is already different